cves/CVE-2014-0231.yml

```diff
@@ -177,38 +206,46 @@ lessons:
     If you think of another lesson we covered in class that applies here, feel
     free to give it a small name and add one in the same format as these.
   defense_in_depth:
-    applies:
-    note:
+    applies: true
+    note: |
+        timeouts are a layer of robustness underneath normal CGI input validation
   least_privilege:
-    applies:
+    applies: false
     note:
   frameworks_are_optional:
-    applies:
+    applies: false
     note:
   native_wrappers:
-    applies:
+    applies: false
     note:
   distrust_input:
-    applies:
-    note:
+    applies: true 
+    note: |
+      This vulnerability is centered around handling harmful input. The 
+      program trusts that it will receive CGI scripts which always terminate.
   security_by_obscurity:
-    applies:
+    applies: false
     note:
   serial_killer:
-    applies:
+    applies: false
     note:
   environment_variables:
-    applies:
+    applies: false
     note:
   secure_by_default:
-    applies:
+    applies: false
     note:
   yagni:
-    applies:
+    applies: false
     note:
   complex_inputs:
-    applies:
-    note:
+    applies: true
+    note: |
+      This vulnerability depends on the fact that CGI script writers may not 
+      anticipate the impact of certain inputs. The complexity of the 
+      relationship between script inputs and execution time means that it may
+      not be apparent to a CGI script writer whether their script will run
+      without halting on all conditions.
```

Comment:
In this case, aren't CGI script writers users of the system? From the perspective of the contributors to httpd, isn't the input just a script that runs for some unknown amount of time? This should be written from the perspective of httpd's contributors.